### PR TITLE
pytorch-lightning - saving model improvements 

### PR DESCRIPTION
**Pull request by Nozdi**

### CODE REVIEW

File: monai_lightning_slurm/run_monai_lightning.py [hunk 1]
```diff
@@ -22,6 +22,9 @@
 PATCH_SIZE = json.loads(os.environ.get('PATCH_SIZE', '[256, 256, 16]'))
 CHANNELS = json.loads(os.environ.get('CHANNELS', '[16, 32, 64]'))
 STRIDES = json.loads(os.environ.get('STRIDES', '[2, 2, 2, 2]'))
+BS = 1
```
Comment (grechut) [inline monai_lightning_slurm/run_monai_lightning.py:25]: Minor: Maybe `BATCH_SIZE` is more solid

File: monai_lightning_slurm/run_monai_lightning.py [hunk 2]
```diff
@@ -213,13 +219,17 @@ def validation_epoch_end(self, outputs):
 #     save_dir=log_dir
 # )
 checkpoint_callback = pytorch_lightning.callbacks.model_checkpoint.ModelCheckpoint(
-    filepath=os.path.join(output_path, "{epoch}-{val_loss:.2f}-{val_dice:.2f}")
+    monitor="val_dice",
+    dirpath=output_path,
+    filename="{epoch}-{val_loss:.2f}-{val_dice:.2f}",
+    save_top_k=3,
```
Comment (grechut) [inline monai_lightning_slurm/run_monai_lightning.py:225]: Minor: Maybe just 1 ? Not sure whether we would ever search among top3, but who knows

File: monai_lightning_slurm/run_monai_lightning.py (hunk 1 above)
Comment (grechut) [inline monai_lightning_slurm/run_monai_lightning.py:25]: Minor: Maybe `BATCH_SIZE` is more solid

Comment (grechut) [inline monai_lightning_slurm/run_monai_lightning.py:25]: Minor: Maybe `BATCH_SIZE` is more solid

File: monai_lightning_slurm/run_monai_lightning.py [hunk 3]
```diff
@@ -149,17 +152,20 @@ def prepare_data(self):
...
-        val_loader = DataLoader(self.val_ds, batch_size=1, num_workers=0)
+        val_loader = DataLoader(self.val_ds, batch_size=BS, num_workers=0)
         return val_loader
 
     def configure_optimizers(self):
-        optimizer = torch.optim.Adam(self._model.parameters(), 1e-3)
-        return optimizer
+        optimizer = torch.optim.Adam(self._model.parameters(), 1e-5)
+        scheduler = torch.optim.lr_scheduler.OneCycleLR(
+            optimizer, max_lr=5e-3, total_steps=NUM_EPOCHS * BS, verbose=True
```
Comment (grechut) [inline monai_lightning_slurm/run_monai_lightning.py:166]: Good idea with setting verbose to True in our early exploration :+1:

File: monai_lightning_slurm/run_monai_lightning.py [hunk 4]
```diff
@@ -235,6 +245,9 @@ def validation_epoch_end(self, outputs):
 
 net.eval()
 device = torch.device("cuda:0")
+net = LymphomaNet.load_from_checkpoint(
+    checkpoint_path=checkpoint_callback.best_model_path
```
Comment (grechut) [inline monai_lightning_slurm/run_monai_lightning.py:249]: Magic ;)

**Review by grechut — APPROVED** [review]
Static code analysis + checking documentations :+1:

Minor thoughts left ;) 

QA when running it in the wild life ;)

File: monai_lightning_slurm/run_monai_lightning.py (hunk 2 above)
Comment (Nozdi) [reply monai_lightning_slurm/run_monai_lightning.py:225]: I was thinking about ensembling 3 best models into one if there are more than 100 epochs diff between them.

Comment (grechut) [reply monai_lightning_slurm/run_monai_lightning.py:225]: Interesting :)

**Review by grechut — APPROVED** [review]

**Pull request merged**